pkg/controller/migplan/migplan_controller.go

```diff
@@ -532,87 +529,100 @@ func (r ReconcileMigPlan) waitForMigAnalyticsReady(plan *migapi.MigPlan) (*migap
 	migAnalytics := &migapi.MigAnalyticList{}
 	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{MigPlan: plan.Name}), migAnalytics)
 	if err != nil {
-		return nil, err
+		return nil, liberr.Wrap(err)
 	}
-	for _, migAnalytic := range migAnalytics.Items {
+	for i := range migAnalytics.Items {
+		migAnalytic := &migAnalytics.Items[i]
 		if migAnalytic.Spec.AnalyzeExtendedPVCapacity == true {
-			if migAnalytic.Status.IsReady() {
-				return &migAnalytic, nil
+			if migAnalytic.Status.IsReady() && !migAnalytic.Spec.Refresh {
+				return migAnalytic, nil
 			}
 			if time.Now().Sub(migAnalytic.CreationTimestamp.Time) > migAnalyticsTimeout {
```

Comment:
This elapsed time measurement seems like it won't work properly for migAnalytics triggered after the initial one, since it uses the creation timestamp.

Reply:
good catch !!

Reply:
@djwhatle Updated to use the `LastTransitionTime` on a Condition set by MigAnalytics indicating that it has started running DF for volumes

Reply:
Makes sense, +1